tests/rptest/services/redpanda.py

```diff
@@ -1986,6 +1987,11 @@ def write_node_conf_file(self,
         # Grab the IP to use it as an alternative listener address, to
         # exercise code paths that deal with multiple listeners
         node_ip = socket.gethostbyname(node.account.hostname)
+        resolved_name = node.account.ssh_output(
+            cmd=f"dig -x {node_ip} +short").decode('utf-8').split(
+                '\n')[0].removesuffix(".")
+        fqdn = node.account.ssh_output(
+            cmd=f"host {resolved_name}").decode('utf-8').split(' ')[0]
```

Comment:
I think it's worth exposing this and reusing it in `RedpandaKerberosNode::_service_principal`

Reply:
Yeah expose something like `RedPandaKerberosNode.get_fqdn(node)`